documents/coredocument.go

```diff
@@ -905,3 +908,32 @@ func (cd *CoreDocument) UnmarshalJSON(data []byte, m Model) error {
 	cd.Document.Attributes, err = toProtocolAttributes(cd.Attributes)
 	return err
 }
+
+// RemoveCollaborators removes DIDs from the Document.
+// Errors out if the document is not in Pending state or collaborators are missing from the document.
+func (cd *CoreDocument) RemoveCollaborators(dids []identity.DID) error {
+	if cd.Status == Committing || cd.Status == Committed {
+		return ErrDocumentNotInAllowedState
+	}
+
+	// remove each collaborator from the roles
+	var found bool
+	for _, did := range dids {
+		for _, role := range cd.Document.Roles {
+			i, f := isDIDInRole(role, did)
+			if !f {
+				continue
+			}
+
+			cd.Modified = true
+			found = true
+			role.Collaborators = byteutils.CutFromSlice(role.Collaborators, i)
+		}
+	}
+
+	if !found {
+		return errors.New("DIDs not found in the Document")
```

Comment:
What do you think about returning a `noop` instead of erroring out? Just a thought.

Reply:
I was thinking that this is a mutated call. If we give noop, we might end up updating the next version with no changes. Hence, I thought we should let user know. Not that I'm thinking, it is a pending doc. Should be okay to return a noop as well